Add crypto, forex, and indices examples

justinpolygon opened the pull request:
Added example code snippets for crypto, forex, and indices.

jbonzo commented on examples/rest/crypto-snapshots_all_tickers.py:
```diff
@@ -0,0 +1,45 @@
+# crunch some numbers
+for item in snapshot:
+
+    # verify this is an TickerSnapshot
+    if isinstance(item, TickerSnapshot):
+
+        # verify this is an Agg
+        if isinstance(item.prev_day, Agg):
+
+            # verify this is a float
+            if isinstance(item.prev_day.open, float) and isinstance(
```
Why do you have to do all this type verification? I believe all of our responses are using type hinting. Do we not have a type hint linter to ensure types are correct?

justinpolygon replied:
Yes, this was to satisfy the `poetry run black --check polygon test_* examples` check that happens. I originally didn't have any of this in here but the linter wanted to enforce all this logic. So, I added it to get the linting to pass.

jbonzo commented on examples/rest/crypto-snapshots_all_tickers.py:
```diff
@@ -0,0 +1,45 @@
+                item.prev_day.close, float
+            ):
+
+                percent_change = (
+                    (item.prev_day.close - item.prev_day.open)
+                    / item.prev_day.open
+                    * 100
+                )
```
Tbh I'm not the biggest fan of this style. If you did this to satisfy a max line number count then I get it. But overall this function is really tall with all the extra whitespace and type checking

justinpolygon replied:
Yeah, I'm not sure why I did that. I'll fix it. Makes sense and agree with you.

justinpolygon replied:
Fixed

justinpolygon replied:
Tagged you in the comment there. Makes sense now that I didn't actually do that and it was the `poetry run black polygon test_* examples` command that's actually reformatting this stuff so the lint checks pass. Not sure what the options are here.

justinpolygon commented:
@jbonzo FYI - it looks like it's actually the linter that made the changes and added the spacing across the lines. So, I changed the code like we chatted about to be:

```
>                 percent_change = ((item.prev_day.close - item.prev_day.open)/item.prev_day.open*100)
>                 print("{:<15}{:<15}{:<15}{:.2f} %".format(item.ticker,item.prev_day.open,item.prev_day.close,percent_change))
```

Then, the link check failed here https://github.com/polygon-io/client-python/actions/runs/4601386532/jobs/8129177080?pr=415 so I run `poetry run black polygon test_* examples` to make the suggested changes and it give me this:

```
<                 percent_change = (
<                     (item.prev_day.close - item.prev_day.open)
<                     / item.prev_day.open
<                     * 100
<                 )
<                 print(
<                     "{:<15}{:<15}{:<15}{:.2f} %".format(
<                         item.ticker,
<                         item.prev_day.open,
<                         item.prev_day.close,
<                         percent_change,
<                     )
<                 )
```

jbonzo commented:
> @jbonzo FYI - it looks like it's actually the linter that made the changes and added the spacing across the lines. So, I changed the code like we chatted about to be:
> 
> 
> 
> ```
> 
> >                 percent_change = ((item.prev_day.close - item.prev_day.open)/item.prev_day.open*100)
> 
> >                 print("{:<15}{:<15}{:<15}{:.2f} %".format(item.ticker,item.prev_day.open,item.prev_day.close,percent_change))
> 
> ```
> 
> 
> 
> Then, the link check failed here https://github.com/polygon-io/client-python/actions/runs/4601386532/jobs/8129177080?pr=415 so I run `poetry run black polygon test_* examples` to make the suggested changes and it give me this:
> 
> 
> 
> ```
> 
> <                 percent_change = (
> 
> <                     (item.prev_day.close - item.prev_day.open)
> 
> <                     / item.prev_day.open
> 
> <                     * 100
> 
> <                 )
> 
> <                 print(
> 
> <                     "{:<15}{:<15}{:<15}{:.2f} %".format(
> 
> <                         item.ticker,
> 
> <                         item.prev_day.open,
> 
> <                         item.prev_day.close,
> 
> <                         percent_change,
> 
> <                     )
> 
> <                 )
> 
> ```

Ah. Thats unfortunate. Ok then.
